File: Courzelo_Back/src/main/java/tn/esprit/courzelo/entities/GroupProject.java
package tn.esprit.courzelo.entities;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "GroupProject")
public class GroupProject {
    @Id
    private long id;
    private String name;
    private int number;
    private List<User_c> students;
    @DBRef
    private Project project;
}

File: Courzelo_Front/Courzelo/src/app/Models/ProjectEntities/GroupProject.ts
import { UserCourzelo } from "../UserCorzelo/UserCourzelo";
import { Project } from "./Project";


export class GroupProject{
    id!:number;
    name!:string;
    Students!:Array<UserCourzelo>;
    project!:Project;
    number!: number;
}